### PR TITLE
Full site link checker

### PR DESCRIPTION
**Pull request by mheap**
### Summary
Add link checker for the latest versions of each product

### Reason
To help us fix all the existing broken links on the docs (for the latest version). We can do a followup if we want to report on _all_ broken links across all versions.

### Testing
The full site scan can only be run on your local machine.

To run a full site scan locally, you'll need the netlify CLI installed.

Do **NOT** run the link checker against production.

To run

```bash
./node_modules/.bin/gulp build
cd dist
netlify dev
```

Then in another terminal:

```bash
cd broken-link-checker
npm ci
node full.js --host http://localhost:8888
```

Finally, be patient. The run will take at least 5 minutes, and up to 30 minutes to complete.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,3 +216,34 @@ The `include_cached` gem uses all passed parameters as the cache lookup key, and
...
+The link checker runs in two different ways:
+
+1. When a pull request is opened, any changed files are detected and those URLs are checked for broken links. This allows us to fix pages incrementally and ensure that we don't break any new links
+1. A full site scan, against the latest version of each product only. This allows us to check all pages for broken links. Once all broken links are fixed, we can retire this job in favour of the CI check
+
+To run a full site scan locally, you'll need the `netlify` CLI installed.
+
+Do **NOT** run the link checker against production.
+
+To run
```
Comment (lena-larionova) [inline README.md:233]:
```suggestion
To run the link checker, build the site locally and start a local Netlify dev environment. From your clone of the doc repo, run:
```

File: README.md [hunk 2]
```diff
@@ -216,3 +216,34 @@ The `include_cached` gem uses all passed parameters as the cache lookup key, and
...
+## Broken links
+
+We check the documentation for broken links using [broken-link-checker](https://github.com/stevenvachon/broken-link-checker) and some custom logic to build a list of excluded URLs.
+
+The link checker runs in two different ways:
+
+1. When a pull request is opened, any changed files are detected and those URLs are checked for broken links. This allows us to fix pages incrementally and ensure that we don't break any new links
+1. A full site scan, against the latest version of each product only. This allows us to check all pages for broken links. Once all broken links are fixed, we can retire this job in favour of the CI check
+
+To run a full site scan locally, you'll need the `netlify` CLI installed.
```
Comment (lena-larionova) [inline README.md:229]:
```suggestion
To run a full site scan locally, you'll need the [`netlify` CLI](https://docs.netlify.com/cli/get-started/) installed.
```

File: README.md [hunk 3]
```diff
@@ -216,3 +216,34 @@ The `include_cached` gem uses all passed parameters as the cache lookup key, and
...
+
+To run
+
+```bash
+./node_modules/.bin/gulp build
+cd dist
+netlify dev
+```
+
+Then in another terminal:
```
Comment (lena-larionova) [inline README.md:241]:
```suggestion
Then in another terminal, from the root of your clone of the docs repo:
```
Or alternatively, add `cd ..` to the commands below.

File: README.md [hunk 4]
```diff
@@ -216,3 +216,34 @@ The `include_cached` gem uses all passed parameters as the cache lookup key, and
...
 For guidelines on how to write includes and call them in target topics, see the
 [Kong Docs contributing guidelines](https://docs.konghq.com/contributing/includes).
+
+## Broken links
+
+We check the documentation for broken links using [broken-link-checker](https://github.com/stevenvachon/broken-link-checker) and some custom logic to build a list of excluded URLs.
+
+The link checker runs in two different ways:
+
+1. When a pull request is opened, any changed files are detected and those URLs are checked for broken links. This allows us to fix pages incrementally and ensure that we don't break any new links
```
Comment (lena-larionova) [inline README.md:226]:
```suggestion
1. When a pull request is opened, any changed files are detected and those URLs are checked for broken links. This allows us to fix pages incrementally and ensure that we don't break any new links.
```

File: README.md [hunk 5]
```diff
@@ -216,3 +216,34 @@ The `include_cached` gem uses all passed parameters as the cache lookup key, and
...
+1. A full site scan, against the latest version of each product only. This allows us to check all pages for broken links. Once all broken links are fixed, we can retire this job in favour of the CI check
```
Comment (lena-larionova) [inline README.md:227]:
```suggestion
1. A full site scan, against the latest version of each product only. This allows us to check all pages for broken links. Once all broken links are fixed, we can retire this job in favour of the CI check.
```

**Review by lena-larionova — COMMENTED** [review]
A thought: can we have verbose output while the link checker is running, or at least one line with something to the effect of "Checking links..."? The current flaw I see is that someone trying to run this will get confused about whether something is happening at all.

I'd also add something to the readme about which error codes are useful (ie, mainly the 404s) and which others can be ignored and bypassed with a label (ie 400s and various unknowns/undefined/notfound).

Or - do we want to leave the label for use for maintainers only? That might be a better idea.

**Comment by mheap** [comment]
Accepted all  code review suggestions + added some output every 1000 links (a full check is approx 250k links at the moment)

I'd advise leaving the label for maintainers only

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**